### PR TITLE
Add Pills Component and Styling

### PR DESCRIPTION
**Pull request by kwood15**
Please can you have a look over this thanks - can't tag until contributor

screen shot 2017-08-21 at 15 56 20

screen shot 2017-08-21 at 15 56 50

### CODE REVIEW

File: demo/src/Components/Pills/Panel.jsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+import T from 'prop-types'
+import classNames from 'classnames'
+
+const Panel = ({ children, className }) =>
+  <div className={classNames('c-pills__panel', className)}>{children}</div>
+
+Panel.propTypes = {
+  className: T.string,
+  children: T.node,
```
Comment (Ehesp) [inline demo/src/Components/Pills/Panel.jsx]: Are children not required here? Or will it just not render and still look ok?

File: demo/src/Components/Pills/Panels.jsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+import T from 'prop-types'
+import classNames from 'classnames'
+
+const Panels = ({ activeIndex, children, className }) => {
+  const filteredChildren = React.Children.map(children, (child, index) => (
+    index === activeIndex ? child : null))
+  return (
```
Comment (Ehesp) [inline demo/src/Components/Pills/Panels.jsx]: If activeIndex is only ever a single integer, could we not do:

```
const child = children[activeIndex] || null;
```

File: demo/src/Components/Pills/Pill.jsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        onClick: handleClick,
+        className: enhancedClassName,
+        ...rest,
+      },
+      children,
+    )
+  )
+}
+
+Pill.propTypes = {
```
Comment (Ehesp) [inline demo/src/Components/Pills/Pill.jsx:61]: What about defaultProps?

File: demo/src/Components/Pills/Pills.jsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+import List from './List'
+import Panels from './Panels'
+
+class Pills extends Component {
+  constructor() {
+    super()
+
+    this.state = {
+      activeIndex: 0,
```
Comment (Ehesp) [inline demo/src/Components/Pills/Pills.jsx:13]: Is there no way to set the default activeIndex to something other than 0?

File: demo/src/Components/Pills/index.jsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+]
+
+const MyPills = () => (
+  <section className="o-section-md">
+    <div className="o-site-wrap o-site-wrap--padding">
+      <h4>Pills</h4>
+      <Pills>
+        <List>
+          {animals.map(({ disabled, name, status }) =>
+            <Pill key={name} disabled={disabled} status={status}>{name}</Pill>)}
```
Comment (Ehesp) [inline demo/src/Components/Pills/index.jsx:53]: Is there a linting enforcement here? usually if props is greater than 1 it would break to new lines

**Review by Ehesp — CHANGES_REQUESTED** [review]

File: demo/src/Components/Pills/index.jsx (hunk 5 above)
Comment (kwood15) [reply demo/src/Components/Pills/index.jsx:53]: Have fixed this across all files - will push separate commits for these

File: demo/src/Components/Pills/Panels.jsx (hunk 2 above)
Comment (kwood15) [reply demo/src/Components/Pills/Panels.jsx]: Done

File: demo/src/Components/Pills/Panel.jsx (hunk 1 above)
Comment (kwood15) [reply demo/src/Components/Pills/Panel.jsx]: Have added required

File: demo/src/Components/Pills/Pill.jsx (hunk 3 above)
Comment (kwood15) [reply demo/src/Components/Pills/Pill.jsx:61]: Have added and removed as we turned this rule off initially

File: demo/src/Components/Pills/Pill.jsx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        to={to}
+        className={enhancedClassName}
+        {...rest}
+      >
+        {children}
+      </ComponentOverride>
+    )
+  }
+
+  return (
```
Comment (rbrtsmith) [inline demo/src/Components/Pills/Pill.jsx:46]: Don't get too hooked up on allowing multiple node types for the demo here, it's just to literally test the styling, these React components here aren't being shipped - It's in Nebula-React where we should care about the React side of things.

**Review by rbrtsmith — COMMENTED** [review]
I wouldn't put unrelated changes into this PR i.e. you are making a lot of changes in the demo that are unrelated to the pills.  It's much easier to manage smaller PRs aimed at a specific purpose.

File: demo/.eslintrc [hunk 7]
```diff
@@ -12,6 +12,8 @@
     "class-methods-use-this": 0,
     "no-plusplus": 0,
     "semi": ["error", "never"],
-    "react/require-default-props": "off"
+    "react/require-default-props": "off",
+    "react/jsx-max-props-per-line" : ['error', {maximum: 1}]
```
Comment (rbrtsmith) [inline demo/.eslintrc]: Not sure I agree with this.  Sometimes we want props on a single line depending on the line length.  Again this is just a demo to test your component so the React code doesn't matter much.  I just pulled in React as a quicker way to test more complex components that had state.

File: demo/src/Components/Pills/Pills.jsx (hunk 4 above)
Comment (rbrtsmith) [reply demo/src/Components/Pills/Pills.jsx:13]: This should be possible, but we only really care about this in Nebula React.  Nebula Components just ships Sass components and Nebula React ships the React components themselves.  The bits of React code are just a dumping ground to test out the components as they are built.  Ideally they can be deleted when build properly in Nebula-React and demoed in http://rbrtsmith.com/nebula/
Hope this makes sense!

Comment (kwood15) [reply demo/src/Components/Pills/Pills.jsx:13]: Yes thanks - I tried this reverting to false and it was breaking so left as is - most of these are fixed now - have left the defaultProp one as have to declare values for each - the only rule I have changed is the one re multiline props so once this gets merged will fixed conflicts on other branch I'm working on.

File: demo/.eslintrc (hunk 7 above)
Comment (kwood15) [reply demo/.eslintrc]: Yes this is why I've only changed this rule as it was raised, have checked all the other components so don't worry

File: demo/src/Components/Pills/Pill.jsx (hunk 6 above)
Comment (kwood15) [reply demo/src/Components/Pills/Pill.jsx:46]: Yes I will do in future Rob - noted

**Comment by kwood15** [comment]
If all ok now could this get merged please cheers @Ehesp

**Comment by rbrtsmith** [comment]
Want that max-props-per-line rule removed from linting prior to merging.

**Comment by rbrtsmith** [comment]
Just noticed are these acting like tabs? If so I should have been more precise in the description that they are just intended to be used for things like tags like the pills on the Bootstrap framework.  
They should also tile... but the styling for that doesn't need to be tied to the pill can just use the matrix list. E.g.
```html
<ul class="o-matrix-list">
  <li class="o-matrix-list__item">
    <button class="c-pill">Pill </button>
  </li>
  <li class="o-matrix-list__item">
    <button class="c-pill">Pill </button>
  </li>
</ul>
```
Thus we dont need the overflow scroll behaviour that the tabs have on smaller screens.

**Comment by kwood15** [comment]
@rbrtsmith - Have removed the multiline prop rule - do you want to raise another story for any new changes for the pills or shall I reacreate another set not tied to tabs? Awaiting this to get merged but if it's not what you want then will leave. Cheers

**Comment by rbrtsmith** [comment]
I would suggest removing the tab based stuff in this branch before merging
then we don't end up with possible redundant code lingering around.

Thanks!

**Pull request closed without merging**